### DISCUSSION TITLE
Start Recording Endpoint cannot find application

### DISCUSSION THREAD

**Discussion by TheTurkeyDev** [OvenMediaLabs/OvenMediaEngine#1507] [Help] [answered]
I'm trying to start recording by making a POST request to `/v1/vhosts/default/apps/live:startRecord` but it returns a 404 with the message `[HTTP] Could not find application: [default/live] (404)`. I've added the FILE publisher to the app and making a GET request to `/v1/vhosts/default/apps/live/streams` returns a 200 and lists the 1 active stream that is going. Am I missing an additional step to enable recording beyond the specified FILE publisher in the server config?

Body of my POST request is :
`
{
    "id": "recorder",
    "stream": {
        "name": "rec"
    },
  "interval": 1800000,
  "segmentationRule": "discontinuity"
}
`

<details>
  <summary>Server.xml</summary>
`
    <?xml version="1.0" encoding="UTF-8" ?>

    <Server version="8">
        <Name>OvenMediaEngine</Name>
        <Type>origin</Type>
        <IP>*</IP>
        <PrivacyProtection>false</PrivacyProtection>
        <StunServer>stun.ovenmediaengine.com:13478</StunServer>
        <Modules>
                
                <HTTP2>
                        <Enable>true</Enable>
                </HTTP2>

                <LLHLS>
                        <Enable>true</Enable>
                </LLHLS>

                
                <P2P>
                        
                        <Enable>false</Enable>
                        <MaxClientPeersPerHostPeer>2</MaxClientPeersPerHostPeer>
                </P2P>
        </Modules>

        
        <Bind>
                
                <Managers>
                        <API>
                                <Port>8081</Port>
                                <WorkerCount>1</WorkerCount>
                        </API>
                </Managers>
                <Providers>
                        
                        <RTSPC>
                                <WorkerCount>1</WorkerCount>
                        </RTSPC>
                        <OVT>
                                <WorkerCount>1</WorkerCount>
                        </OVT>
                        
                        <RTMP>
                                <Port>${env:OME_RTMP_PROV_PORT:1935}</Port>
                                <WorkerCount>1</WorkerCount>
                        </RTMP>
                        <SRT>
                                <Port>${env:OME_SRT_PROV_PORT:9999}</Port>
                                <WorkerCount>1</WorkerCount>
                        </SRT>
                        
                        <WebRTC>
                                <Signalling>
                                        <Port>${env:OME_WEBRTC_SIGNALLING_PORT:3333}</Port>
                                        <TLSPort>${env:OME_WEBRTC_SIGNALLING_TLS_PORT:3334}</TLSPort>
                                        <WorkerCount>1</WorkerCount>
                                </Signalling>
                                <IceCandidates>
                                        <IceCandidate>${env:OME_HOST_IP:*}:${env:OME_WEBRTC_CANDIDATE_PORT:10000-10004/udp}</IceCandidate>
                                        <TcpRelay>${env:OME_HOST_IP:*}:${env:OME_WEBRTC_TCP_RELAY_PORT:3478}</TcpRelay>
                                        <TcpForce>true</TcpForce>
                                        <TcpRelayWorkerCount>1</TcpRelayWorkerCount>
                                </IceCandidates>
                        </WebRTC>
                </Providers>

                <Publishers>
                        
                        <OVT>
                                <Port>${env:OME_ORIGIN_PORT:9000}</Port>
                                <WorkerCount>1</WorkerCount>
                        </OVT>
                        <LLHLS>
                                
                                <Port>${env:OME_LLHLS_STREAM_PORT:3333}</Port>
                                <TLSPort>${env:OME_LLHLS_STREAM_TLS_PORT:3334}</TLSPort>
                                <WorkerCount>1</WorkerCount>
                        </LLHLS>
                        <WebRTC>
                                <Signalling>
                                        <Port>${env:OME_WEBRTC_SIGNALLING_PORT:3333}</Port>
                                        <TLSPort>${env:OME_WEBRTC_SIGNALLING_TLS_PORT:3334}</TLSPort>
                                        <WorkerCount>1</WorkerCount>
                                </Signalling>
                                <IceCandidates>
                                        <IceCandidate>${env:OME_WEBRTC_CANDIDATE_IP:*}:${env:OME_WEBRTC_CANDIDATE_PORT:10000-10004/udp}</IceCandidate>
                                        <TcpRelay>${env:OME_WEBRTC_CANDIDATE_IP:*}:${env:OME_WEBRTC_TCP_RELAY_PORT:3478}</TcpRelay>
                                        <TcpForce>true</TcpForce>
                                        <TcpRelayWorkerCount>1</TcpRelayWorkerCount>
                                </IceCandidates>
                        </WebRTC>
                        <Thumbnail>
                                <Port>20080</Port>
                        </Thumbnail>
                </Publishers>
        </Bind>
        <Managers>
                <API>
                        <AccessToken>██████</AccessToken>
                        <CrossDomains>
                                <Url>*</Url>
                        </CrossDomains>
                </API>
        </Managers>
        <VirtualHosts>
                
                <VirtualHost include="VHost*.xml" />
                <VirtualHost>
                        <Name>default</Name>
                        
                        <Distribution>ovenmediaengine.com</Distribution>

                        
                        <Host>
                                <Names>
                                        
                                        <Name>*</Name>
                                </Names>
                                <TLS>
                                        <CertPath>./cert.crt</CertPath>
                                        <KeyPath>./cert.key</KeyPath>
                                        <ChainCertPath>./cert.ca-bundle</ChainCertPath>
                                </TLS>
                        </Host>
                        <AdmissionWebhooks>
                                <ControlServerUrl>http://██████/rtmp/on-publish</ControlServerUrl>
                                <SecretKey>1234</SecretKey>
                                <Timeout>3000</Timeout>
                                <Enables>
                                        <Providers>rtmp</Providers>
                                        <Publishers></Publishers>
                                </Enables>
                        </AdmissionWebhooks>

                        

                        

                        
                        <CrossDomains>
                                <Url>*</Url>
                        </CrossDomains>

                        
                        <Applications>
                                <Application>
                                        <Name>live</Name>
                                        
                                        <Type>live</Type>
                                        <OutputProfiles>
                                                
                                                

                                                
                                                <HWAccels>
                                                        <Decoder>
                                                                        <Enable>false</Enable>
                                                                        
                                                                        
                                                        </Decoder>
                                                        <Encoder>
                                                                        <Enable>false</Enable>
                                                                        
                                                        </Encoder>
                                                </HWAccels>

                                                <OutputProfile>
                                                        <Name>bypass_stream</Name>
                                                        <OutputStreamName>${OriginStreamName}</OutputStreamName>

                                                        

                                                        <Encodes>
                                                                <Video>
                                                                        <Name>bypass_video</Name>
                                                                        <Bypass>true</Bypass>
                                                                </Video>
                                                                <Audio>
                                                                        <Name>aac_audio</Name>
                                                                        <Codec>aac</Codec>
                                                                        <Bitrate>128000</Bitrate>
                                                                        <Samplerate>48000</Samplerate>
                                                                        <Channel>2</Channel>
                                                                        <BypassIfMatch>
                                                                                <Codec>eq</Codec>
                                                                        </BypassIfMatch>
                                                                </Audio>
                                                                <Audio>
                                                                        <Name>opus_audio</Name>
                                                                        <Codec>opus</Codec>
                                                                        <Bitrate>128000</Bitrate>
                                                                        <Samplerate>48000</Samplerate>
                                                                        <Channel>2</Channel>
                                                                        <BypassIfMatch>
                                                                                <Codec>eq</Codec>
                                                                        </BypassIfMatch>
                                                                </Audio>
                                                                <Image>
                                                                        <Codec>jpeg</Codec>
                                                                        <Framerate>1</Framerate>
                                                                        <Width>1280</Width>
                                                                        <Height>720</Height>
                                                                </Image>
                                                                <Image>
                                                                        <Codec>png</Codec>
                                                                        <Framerate>1</Framerate>
                                                                        <Width>1280</Width>
                                                                        <Height>720</Height>
                                                                </Image>
                                                                
                                                        </Encodes>
                                                </OutputProfile>
                                        </OutputProfiles>
                                        <Providers>
                                                <OVT />
                                                <WebRTC>
                                                        <Timeout>30000</Timeout>
                                                        <CrossDomains>
                                                                <Url>*</Url>
                                                        </CrossDomains>
                                                </WebRTC>
                                                <RTMP />
                                                <SRT />
                                                <RTSPPull />
                                                <Schedule>
                                                        <MediaRootDir>/opt/ovenmediaengine/media</MediaRootDir>
                                                        <ScheduleFilesDir>/opt/ovenmediaengine/media</ScheduleFilesDir>
                                                </Schedule>
                                                
                                        </Providers>
                                        <Publishers>
                                                <AppWorkerCount>1</AppWorkerCount>
                                                <StreamWorkerCount>8</StreamWorkerCount>
                                                <OVT />
                                                <WebRTC>
                                                        <Timeout>30000</Timeout>
                                                        <Rtx>false</Rtx>
                                                        <Ulpfec>false</Ulpfec>
                                                        <JitterBuffer>false</JitterBuffer>
                                                </WebRTC>
                                                <LLHLS>
                                                        <OriginMode>true</OriginMode>
                                                        
                                                        <ChunkDuration>0.5</ChunkDuration>
                                                        
                                                        <PartHoldBack>1.5</PartHoldBack>
                                                        <SegmentDuration>6</SegmentDuration>
                                                        <SegmentCount>10</SegmentCount>
                                                        <CrossDomains>
                                                                <Url>*</Url>
                                                        </CrossDomains>
                                                </LLHLS>
                                                <Thumbnail>
                                                        <CrossDomains>
                                                                <Url>*</Url>
                                                        </CrossDomains>
                                                </Thumbnail>
                                                <FILE>
                                                        <RootPath>/videos</RootPath>
                                                        <FilePath>/${VirtualHost}/${Application}/${Stream}/${StartTime:YYYYMMDDhhmmss}_${EndTime:YYYYMMDDhhmmss}.ts</FilePath>
                                                        <InfoPath>/${VirtualHost}/${Application}/${Stream}.xml</InfoPath>
                                                </FILE>
                                        </Publishers>
                                </Application>
                        </Applications>
                </VirtualHost>
        </VirtualHosts>
</Server>
`
</details>

Thanks!

**Comment by Keukhan** [comment]
@TheTurkeyDev 

Thanks for reporting.
Detailed information is needed for cause analysis.
Is the [**live**] application defined in server.xml settings?

**Reply by TheTurkeyDev** [reply]
It is. I've added my Sever.xml to the description above

**Reply by TheTurkeyDev** [reply]
The server restarted over night and now the POST request goes through with no issue. I swear that I was restarting the OME docker container after changing the Server.xml file, but maybe something wasn't reloading correctly? Not sure, but either way it seems to be working now

**Reply by TheTurkeyDev** [reply]
This being said, the Recorder doesn't seem to be starting. I've restarted the stream, but the recorders state just stays in the ready state. I am using AdmissionWebhooks for validation and the stream get's redirected to the live app from this. Could that cause an issue?

**Reply by bchah** [reply]
3 ideas to try:

1) Let's see the OME logs around the time of that request - they will provide more clues
2) Does OME have appropriate permissions to write to that directory "/videos" ?
3) Try specifying the tracks (variantNames) in your recording request, specifically bypass_video and aac_audio for easy remuxing. You have two image encoders going which by the way will consume a lot of CPU.

**Reply by TheTurkeyDev** [reply]
1. Here are the logs when a stream connects

<details>
  <summary>Logs</summary>

```
[2024-01-24 16:34:24.138] I [SPRTMP-t1935:26] Provider | stream.cpp:51   | Unknown/(103) has been started stream
[2024-01-24 16:34:24.138] I [SPRTMP-t1935:26] RTMPProvider | rtmp_provider.cpp:189  | A RTMP client has connected from <ClientSocket: 0x7efcee402410, #31, Connected, TCP, Nonblocking, 192.168.1.244:43100>
[2024-01-24 16:34:24.160] I [SPRTMP-t1935:26] AccessController | access_controller.cpp:257  | AdmissionWebhooks queried http://192.168.1.200:8091/rtmp/on-publish whether client 192.168.1.244:43100 could access rtmp://192.168.1.200:1935/XVlBzgbaiCMRAjWwhTHctcuAxhxKQFDa/. (Result : Allow Elapsed : 0 ms)
[2024-01-24 16:34:24.183] I [SPRTMP-t1935:26] MediaRouter | mediarouter_application.cpp:340  | [#default#live/2(103)] Trying to create a stream
[2024-01-24 16:34:24.183] I [SPRTMP-t1935:26] Monitor | application_metrics.cpp:58   | Create StreamMetrics(2/334d2520-d89f-4ac9-8609-369fe335943e/default/#default#live/2/i) for monitoring
[2024-01-24 16:34:24.183] I [SPRTMP-t1935:26] MediaRouter | mediarouter_application.cpp:440  | [#default#live/2(103)] Stream has been created
[2024-01-24 16:34:24.183] I [SPRTMP-t1935:26] Transcoder | transcoder_stream.cpp:63   | [#default#live/2(103)] Transcoder stream has been started
[2024-01-24 16:34:24.183] I [InboundWorker:29] MediaRouter | mediarouter_application.cpp:477  | [#default#live/2(103)] Stream has been prepared 
[Stream Info]
id(103), msid(0), output(2), SourceType(Rtmp), RepresentationType(Source), Created Time (Wed Jan 24 16:34:24 2024) UUID(334d2520-d89f-4ac9-8609-369fe335943e/default/#default#live/2/i)
	Video Track #0: Public Name(Video_0) Variant Name(Video) Bitrate(0b) Codec(1,H264,none:0) BSF(AVCC) Resolution(1920x1080) Framerate(50.00) KeyInterval(0/frame) SkipFrames(0) BFrames(0) timebase(1/1000)
	Audio Track #1: Public Name(Audio_1) Variant Name(Audio) Bitrate(125.00Kb) Codec(6,AAC,none) BSF(AAC_RAW) Samplerate(44.1K) Format(s16, 16) Channel(stereo, 2) timebase(1/1000)
	Data  Track #2: Public Name(Data_2) Variant Name(Data) Codec(0,Unknown,none) BSF(ID3v2) timebase(1/1000)
[2024-01-24 16:34:24.188] I [InboundWorker:29] Transcoder | transcoder_stream.cpp:128  | [#default#live/2(103)] Using local output profiles by webhook
[2024-01-24 16:34:24.189] I [InboundWorker:29] Transcoder | transcoder_stream.cpp:410  | [#default#live/2(103)] Output stream has been created. [#default#live/2(1517684406)]
[2024-01-24 16:34:24.189] I [InboundWorker:29] Transcoder | transcoder_decoder.cpp:242  | The decoder has been created successfully. track(#0) codec(H264), module(default:0)
[2024-01-24 16:34:24.189] I [InboundWorker:29] Transcoder | transcoder_decoder.cpp:242  | The decoder has been created successfully. track(#1) codec(AAC), module(default:0)
[2024-01-24 16:34:24.189] I [InboundWorker:29] MediaRouter | mediarouter_application.cpp:340  | [#default#live/2(1517684406)] Trying to create a stream
[2024-01-24 16:34:24.189] I [InboundWorker:29] Monitor | application_metrics.cpp:58   | Create StreamMetrics(2/334d2520-d89f-4ac9-8609-369fe335943e/default/#default#live/2/o) for monitoring
[2024-01-24 16:34:24.189] I [InboundWorker:29] MediaRouter | mediarouter_application.cpp:440  | [#default#live/2(1517684406)] Stream has been created
[2024-01-24 16:34:24.189] I [InboundWorker:29] Transcoder | transcoder_stream.cpp:150  | [#default#live/2(103)] Transcoder stream has been prepared
[2024-01-24 16:34:24.189] I [Decaac:247] Transcoder | decoder_aac.cpp:203  | [#default#live/2(103)] input track information: [audio] aac (LC), 44100 Hz, stereo(2), fltp, 10 kbps, timebase: 1/1000, 
[2024-01-24 16:34:24.189] I [Decaac:247] Transcoder | transcoder_encoder.cpp:278  | The encoder has been created successfully. track(#4), codec(OPUS), module(default:0)
[2024-01-24 16:34:24.190] I [Decaac:247] Transcoder | filter_resampler.cpp:159  | Resampler parameters. track(#1 -> #4). desc(src: time_base=1/1000:sample_rate=44100:sample_fmt=fltp:channel_layout=stereo -> output: asettb=1/48000,aresample=async=1000,aresample=48000,aformat=sample_fmts=s16:channel_layouts=stereo,asetnsamples=n=960)
[2024-01-24 16:34:24.208] I [Dech264:246] Transcoder | decoder_avc.cpp:210  | [#default#live/2(103)] input track information: [video] h264 (High 4.2), yuv420p, 1920x1080 [SAR 0:1 DAR 16:9], 50 fps, 195 kbps, timebase: 1/100, 
[2024-01-24 16:34:24.208] I [Dech264:246] Transcoder | transcoder_encoder.cpp:278  | The encoder has been created successfully. track(#1), codec(JPEG), module(default:0)
[2024-01-24 16:34:24.208] I [Dech264:246] Transcoder | transcoder_encoder.cpp:278  | The encoder has been created successfully. track(#2), codec(PNG), module(default:0)
[2024-01-24 16:34:24.208] I [Dech264:246] Transcoder | filter_rescaler.cpp:298  | Rescaler parameters. track(#0 -> #1), module(default:0 -> default:0). desc(src:video_size=1920x1080:pix_fmt=yuv420p:time_base=1/1000:pixel_aspect=1/1:frame_rate=30/1 -> output:fps=fps=1.00:round=near,settb=1/90000,scale=1280x720:flags=bilinear,format=yuv420p)
[2024-01-24 16:34:24.210] I [Dech264:246] Transcoder | filter_rescaler.cpp:298  | Rescaler parameters. track(#0 -> #2), module(default:0 -> default:0). desc(src:video_size=1920x1080:pix_fmt=yuv420p:time_base=1/1000:pixel_aspect=1/1:frame_rate=30/1 -> output:fps=fps=1.00:round=near,settb=1/90000,scale=1280x720:flags=bilinear,format=rgba)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] MediaRouter | mediarouter_application.cpp:477  | [#default#live/2(1517684406)] Stream has been prepared 
[Stream Info]
id(1517684406), msid(0), output(2), SourceType(Transcoder), RepresentationType(Source), Created Time (Wed Jan 24 16:34:24 2024) UUID(334d2520-d89f-4ac9-8609-369fe335943e/default/#default#live/2/o)
	>> Origin Stream Info
	id(103), output(2), SourceType(Rtmp), Created Time (Wed Jan 24 16:34:24 2024)
	Video Track #0: Public Name(Video_0) Variant Name(bypass_video) Bitrate(10.16Mb) Codec(1,H264,Passthrough:0) BSF(AVCC) Resolution(1920x1080) Framerate(50.00) KeyInterval(0/frame) SkipFrames(0) BFrames(0) timebase(1/1000)
	Video Track #1: Public Name(Video_0) Variant Name(image_0) Bitrate(1.00Mb) Codec(9,JPEG,default:0) BSF(JPEG) Resolution(1280x720) Framerate(1.00) KeyInterval(0/frame) SkipFrames(0) BFrames(0) timebase(1/90000)
	Video Track #2: Public Name(Video_0) Variant Name(image_1) Bitrate(1.00Mb) Codec(10,PNG,default:0) BSF(PNG) Resolution(1280x720) Framerate(1.00) KeyInterval(0/frame) SkipFrames(0) BFrames(0) timebase(1/90000)
	Audio Track #3: Public Name(Audio_1) Variant Name(aac_audio) Bitrate(128.00Kb) Codec(6,AAC,Passthrough) BSF(AAC_RAW) Samplerate(44.1K) Format(fltp, 32) Channel(stereo, 2) timebase(1/1000)
	Audio Track #4: Public Name(Audio_1) Variant Name(opus_audio) Bitrate(128.00Kb) Codec(8,OPUS,default) BSF(OPUS) Samplerate(48.0K) Format(s16, 16) Channel(stereo, 2) timebase(1/48000)
	Data  Track #5: Public Name(Data_2) Variant Name(Data) Codec(0,Unknown,Passthrough) BSF(ID3v2) timebase(1/1000)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] WebRTC Publisher | rtc_stream.cpp:163  | RtcStream(#default#live/2) - Ignore unsupported codec(JPEG)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] WebRTC Publisher | rtc_stream.cpp:163  | RtcStream(#default#live/2) - Ignore unsupported codec(PNG)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] WebRTC Publisher | rtc_stream.cpp:163  | RtcStream(#default#live/2) - Ignore unsupported codec(AAC)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] WebRTC Publisher | rtc_stream.cpp:200  | WebRTC Stream has been created : 2/1517684406
Rtx(false) Ulpfec(false) JitterBuffer(false) PlayoutDelay(false min:0 max: 0)
[2024-01-24 16:34:25.228] I [OutboundWorker:30] Publisher | stream.cpp:212  | WebRTC Publisher Application application has started [2(1517684406)] stream (MSID : 0)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] LLHLS Publisher | llhls_stream.cpp:118  | LLHlsStream(#default#live/2) - Ignore unsupported codec(JPEG)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] LLHLS Publisher | llhls_stream.cpp:118  | LLHlsStream(#default#live/2) - Ignore unsupported codec(PNG)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] LLHLS Publisher | llhls_stream.cpp:118  | LLHlsStream(#default#live/2) - Ignore unsupported codec(OPUS)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] LLHLS Publisher | llhls_stream.cpp:213  | LLHlsStream has been created : 2/1517684406
OriginMode(true) Chunk Duration(0.50) Segment Duration(6) Segment Count(10) DRM(none)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] Publisher | stream.cpp:212  | LLHLS Publisher Application application has started [2(1517684406)] stream (MSID : 0)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] Publisher | stream.cpp:212  | OVTPublisher Application application has started [2(1517684406)] stream (MSID : 0)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] Publisher | stream.cpp:212  | FilePublisher Application application has started [2(1517684406)] stream (MSID : 0)
[2024-01-24 16:34:25.229] I [OutboundWorker:30] Publisher | stream.cpp:212  | ThumbnailPublisher Application application has started [2(1517684406)] stream (MSID : 0)
```
</details>

2. It should have permissions. I can make the folder accessible for any user in the meantime to rule that out
3. Oh really? It's just for thumbnails that I grabbed from the documentation. Is there a better/ more efficient way to do this?

**Reply by bchah** [reply]
Those logs show the stream starting up OK - how about around the time of the recording request? It could be struggling to record the 6 tracks that comprise the stream - you should specify just the video and audio variants you want recorded in the API request. Logs will tell all (I hope!)

Thumbnail-wise, OME constantly decodes the stream whether you are fetching a thumbnail or not. Streams of very high bitrate + resolution or with advanced codecs / colour formats can cause OME to crash when encoders are active due to a memory leak that occurs when the single-threaded Dech26x process is overwhelmed. There's an interesting workaround in #1447 but if you aren't running a lot of streams or advanced video formats you should be OK with a few thumbnail publishers running.

**Reply by TheTurkeyDev** [reply]
Oh wow, yeah those 2 image decoders were chewing up ~10% CPU. For now I've commented them out and set the `variantNames` on the request to only `bypass_video` and `aac_audio`, but still am not seeing any recording files generated and the state remains `ready`. Checking the logs after making the POST request for `startRecord` shows no new log lines added

**Reply by bchah** [reply]
Check your Logger.xml - set "FILE" tags to "debug" (or everything for simplicity)

`<Tag name="FILE" level="debug" />`
`<Tag name=".*" level="debug" />`

**Reply by TheTurkeyDev** [reply]
Still no new logs when recording get's enabled. When I toggled on debug for everything I did see a lot of logs containing ffmpeg, so the log level was changed

**Reply by bchah** [reply]
That implies that your request isn't reaching the server at all 😳 

When you say "the POST goes through" that means you should be receiving a 200 OK response and there whould be HTTP.Server log entries at least reflecting that response. Anything there? 

And if you GET /v1/vhosts/default/apps/live:records does the recording you started show up in the response?

**Reply by TheTurkeyDev** [reply]
Correct I receive a 200 back ie
`
{
    "message": "OK",
    "response": {
        "app": "live",
        "createdTime": "2024-01-24T18:41:56.213+00:00",
        "id": "recorder",
        "interval": 1800000,
        "segmentationRule": "discontinuity",
        "state": "ready",
        "stream": {
            "name": "rec",
            "trackIds": [],
            "variantNames": [
                "bypass_video",
                "aac_audio"
            ]
        },
        "vhost": "default"
    },
    "statusCode": 200
}
`
And calling records also show the above.
I just tested this with a fresh instance (I actually tried out the OME docker launcher) and only added the API & recording info and I got the same results. No logs when hitting the API, but I get a 200 ok response back with the expected data. Recording does not start and calling `/v1/vhosts/default/apps/app:records` (app because I didn't change it to live) returns the expected data, but the state is only "ready". Calling `/v1/vhosts/default/apps/app/streams` returns  `{
    "message": "OK",
    "response": [
        "test"
    ],
    "statusCode": 200
}` Which makes me think that it should be recording no?

**Reply by bchah** [reply, marked as answer]
It should be recording, yes - but it looks like it's ready for a stream called "rec" where your streams endpoint is returning a stream called "test". Can you give that a check?

**Reply by TheTurkeyDev** [reply]
🤦‍♂️I'm so sorry. Yes, the stream was wrong. Changing that to the correct stream name fixed it. Thank you for being so patient and for letting me know about the thumbnail recordings. Appreciate all the help and to Airensoft* for making this amazing engine!

**Reply by bchah** [reply]
Glad we found it and happy to help! I am just a user like you, the real credit goes to Airensoft for making OvenMediaEngine available to us at no cost 🥇

**Reply by naanlizard** [reply]
They are very good devs, I'm sure they'd appreciate some monetary support too if you're able. Open source devs also need to eat :)

https://github.com/sponsors/AirenSoft
